### PR TITLE
Integrate with application insights

### PR DESCRIPTION
**Pull request by c-w**
We make use of three application insights apis:

- `trackTrace` for log statements
- `trackDependency` for call to third-party services
- `trackEvent` for first-party scenarios

### CODE REVIEW

File: src/clients/appinsights/AppInsightsClient.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  if (!client) return promiseFunc;
+
+  function eventTracker(...args) {
+    return new Promise((resolve, reject) => {
+      const start = new Date();
+      promiseFunc(...args)
+      .then(returnValue => {
+        const duration = new Date() - start;
+        const success = true;
+        client.trackEvent(eventName, { duration: duration, success: success });
```
Comment (Smarker) [inline src/clients/appinsights/AppInsightsClient.js]: What if you wanted to add other properties to the event like siteType? Or metrics like the number of inserts performed? Would it be more flexible to add a third optional parameter to `trackEvent` for properties and a fourth optional parameter to it for metrics?

File: src/clients/appinsights/AppInsightsClient.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+let consoleError = console.error;
+let consoleWarn = console.warn;
+
+function setup() {
+  if (appInsightsKey) {
+    const appInsights = require('applicationinsights');
+    appInsights.setup(appInsightsKey);
+    appInsights.start();
+    client = appInsights.getClient(appInsightsKey);
+    console.log = trackTrace(/* INFO */ 1, consoleLog);
```
Comment (Smarker) [inline src/clients/appinsights/AppInsightsClient.js]: Would enumerating the values 1, 2, 3 as INFO, ERROR, WARNING be more clear?

**Review by Smarker — APPROVED** [review]
LGTM with comments

File: src/clients/appinsights/AppInsightsClient.js (hunk 2 above)
Comment (c-w) [reply src/clients/appinsights/AppInsightsClient.js]: Yes! ~~7ef54c3~~ a9f93b0.

File: src/clients/appinsights/AppInsightsClient.js (hunk 1 above)
Comment (c-w) [reply src/clients/appinsights/AppInsightsClient.js]: Done in ~~8f9e920~~ 79fd918.

**Pull request merged**